cli/exp_task_create.go

```diff
@@ -173,7 +199,107 @@ func (r *RootCmd) taskCreate() *serpent.Command {
 				cliui.Timestamp(task.CreatedAt),
 			)
 
-			return nil
+			if !wait {
+				return nil
+			}
+
+			var (
+				eg   errgroup.Group
```

Comment:
Suggestion: Consider using `ctx, eg := errgroup.WithContext(ctx)` to allow ctx closure on errgroup failure.

Also consider initializing `waitCtx` earlier and making it parent to the errgroup context so that it propagates to both routines. If the first group always returned an error, the `done` channel could probably be omitted as well.